File: python/sparklabs/batch/sales/dataset/customer_orders.py
from pyspark.sql import functions as F

from sparklabs.batch.sales.dataset import utils


def run() -> None:
    (
        utils.customers.join(utils.orders, ["customer_id"], "inner")
        .groupBy(F.col("customer_id"))
        .agg(F.approx_count_distinct("customer_id").alias("total_orders"))
        .orderBy(F.desc("total_orders"))
        .show(truncate=False)
    )


if __name__ == "__main__":
    run()

File: r/sparklabs/batch/sales/dataframe/customer_orders.R
source("r/sparklabs/batch/sales/dataframe/utils.R")

result <- orderBy(
  agg(
    groupBy(join(customers, orders, customers$customer_id == orders$customer_id), customers$customer_id),
    total_orders = approx_count_distinct(customers$customer_id)
  ),
  desc("total_orders")
)
showDF(result, truncate = FALSE)
